docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1345,444 +1370,119 @@ To create, accept, or cancel escrows, it's better to implement mutations for eac
 
 ##### `LockedObject` and `Locked` component
 
 The `<LockedObject />` (`src/components/locked/LockedObject.tsx`) component is mainly responsible for mapping an on-chain `SuiObjectData` `Locked` object to its corresponding `ApiLockedObject`, which is finally delegated to the `<Locked />` component for rendering. The `<LockedObject />` fetches the locked item object ID if the prop `itemId` is not supplied by using `dApp Kit` `useSuiClientQuery()` hook to call the `getDynamicFieldObject` RPC endpoint. Recalling that in this smart contract, the locked item is put into a dynamic object field.
 
+{@inject: examples/trading/frontend/src/components/locked/LockedObject.tsx}
+
 The `<Locked />` (`src/components/locked/partials/Locked.tsx`) component is mainly responsible for rendering the `ApiLockedObject`. It also consists of several on-chain interactions: unlock the locked objects and create an escrow out of the locked object.
 
-```ts title='src/components/locked/LockedObject.tsx'
-/**
- * Acts as a wrapper between the `Locked` object fetched from API
- * and the on-chain object state.
- *
- * Accepts an `object` of type `::locked::Locked`, fetches the itemID (though the DOF)
- * and then renders the `Locked` component.
- *
- * ItemId is optional because we trust the API to return the correct itemId for each Locked.
- */
-export function LockedObject({
-	object,
-	itemId,
-	hideControls,
-}: {
-	object: SuiObjectData;
-	itemId?: string;
-	hideControls?: boolean;
-}) {
-	const owner = () => {
-		if (!object.owner || typeof object.owner === 'string' || !('AddressOwner' in object.owner))
-			return undefined;
-		return object.owner.AddressOwner;
-	};
+{@inject: examples/trading/frontend/src/components/locked/partials/Locked.tsx}
 
-	const getKeyId = (item: SuiObjectData) => {
-		if (!(item.content?.dataType === 'moveObject') || !('key' in item.content.fields)) return '';
-		return item.content.fields.key as string;
-	};
+The `Locked` component is a bit complex because it has multiple actions that can be performed on the locked object. The `Locked` component is responsible for rendering the `ApiLockedObject` and providing the necessary actions to interact with the locked object. The `Locked` component consists of the following actions:
```

Comment:
This is where the value of a guide comes in. For the `Locked` component, is it possible for us to build up the component feature-by-feature, so that instead of explaining its (complex) end state in one go, we start off with a simple component and layer on complexity?

For example, we start off just with an explorer link, then we add the `useSuiClientQuery` call so we get the Display information and render something from that, then the "UnlocK" button, then the "Create Escrow" button.